oauth/random.go

```diff
@@ -1,18 +1,39 @@
 package oauth
 
 import (
-	"math/rand"
-	"time"
+	crand "crypto/rand"
+	"encoding/binary"
+	"log"
+	rand "math/rand"
 )
 
 const charset = "abcdefghijklmnopqrstuvwxyz" +
 	"ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
 
+var src cryptoSource
+
 func GenerateRandomString(length int) string {
-	rand.Seed(time.Now().UnixNano())
+	rnd := rand.New(src)
 	b := make([]byte, length)
 	for i := range b {
-		b[i] = charset[rand.Intn(len(charset))]
+		b[i] = charset[rnd.Intn(len(charset))]
 	}
 	return string(b)
 }
+
+type cryptoSource struct{}
+
+func (s cryptoSource) Seed(seed int64) {}
+
+func (s cryptoSource) Int63() int64 {
+	return int64(s.Uint64() & ^uint64(1<<63))
+}
+
+func (s cryptoSource) Uint64() (v uint64) {
+	err := binary.Read(crand.Reader, binary.BigEndian, &v)
```

Comment:
Is the thought that the standard seeded random is deterministic and its possible to spoof? I don't have much experience in this area unfortunately :(

Reply:
Yes. This is the idea behind this. 

As in the previous implementation of this.....Generally using the time as a seed is not a good idea. Each time you set the same seed, you get the same sequence. So if you're setting the seed to the time in a fast loop, you'll probably call it with the same seed many times. That's not good for supposed "secure" tokens.

Reply:
math/rand is deterministic by default.... [example](https://github.com/golang/go/issues/4509)